assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -246,7 +247,14 @@ pub fn u32rotr(span_builder: &mut BasicBlockBuilder, imm: Option<u8>) -> Result<
             span_builder.push_op(Noop);
         },
         Some(imm) => {
-            validate_param(imm, 1..=MAX_U32_ROTATE_VALUE)?;
+            if imm > MAX_U32_ROTATE_VALUE {
+                return Err(AssemblyError::InvalidU8Param {
+                    span: SourceSpan::UNKNOWN,
+                    source_file: None,
```

Comment:
Similarly here, we should get these from the `ProcedureContext` (that would be passed into this function)